### PR TITLE
Reorganization of the "Configuring Che" chapter.

### PR DESCRIPTION
**Pull request by themr0c**
Reorganise the "Configuring Che" chapter.

* Emphasis on editing the `CheCluster` Custom Resource as default configuration method.
* Regroup all sections in one place.

### CODE REVIEW

File: src/main/pages/che-7/installation-guide/assembly_configuring-che-overview.adoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+[id="configuring-che-overview_{context}"]
+= Configuring Che overview
+
+:context: configuring-che-overview
+
+The following chapter describes configuration methods and options for {prod}, with some user stories as example.
+
+The two first sections describe the methods available to configure {prod}.
+
+* link:{site-baseurl}che-7/configuring-the-che-installation[Configuring the {prod-short} installation] describes configuration options to install {prod} using the Operator.
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/assembly_configuring-che-overview.adoc:23]: I imagine that, in a future PR, in the installation parts (which is currently in *QuickStarts*), you would also have a sort of introduction that lists the various installation methods and point out those that use the Operator and should thus be configured though a `CheCluster`.

File: src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[id="understanding-the-checluster-custom-resource_{context}"]
+= Understanding the `CheCluster` Custom Resource
+
+A default deployment of {prod-short} consist in the application of a parametrized `CheCluster` link:https://docs.openshift.com/container-platform/latest/operators/crds/crd-managing-resources-from-crds.html[Custom Resource] by the {prod} link:https://docs.openshift.com/container-platform/latest/operators/olm-what-operators-are.html[Operator].
+The `CheCluster` Custom Resource is a YAML document describing the configuration of the {prod-short} cluster.
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc]:
```suggestion
The `CheCluster` Custom Resource is a YAML document describing the configuration of the overall {prod-short} installation.
```

File: src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+The Operator translates the `CheCluster` Custom Resource into configuration usable by each component: authentication, database, registry, server, storage.
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc]:
```suggestion
The Operator translates the `CheCluster` Custom Resource into configuration usable by each component of the {prod-short} installation: authentication, database, registry, server, storage. 
```

File: src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc [hunk 4]
```diff
@@ -0,0 +1,18 @@
+[id="understanding-the-checluster-custom-resource_{context}"]
+= Understanding the `CheCluster` Custom Resource
+
+A default deployment of {prod-short} consist in the application of a parametrized `CheCluster` link:https://docs.openshift.com/container-platform/latest/operators/crds/crd-managing-resources-from-crds.html[Custom Resource] by the {prod} link:https://docs.openshift.com/container-platform/latest/operators/olm-what-operators-are.html[Operator].
+The `CheCluster` Custom Resource is a YAML document describing the configuration of the {prod-short} cluster.
+The Operator translates the `CheCluster` Custom Resource into configuration usable by each component: authentication, database, registry, server, storage. 
+
+For instance, to configure the main properties of the {prod-short} server component, the Operator generates the necessary `configMap`, called `che`. Any change in the `configMap` automatically triggers a restart of the {prod-short} Pod.
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc]:
```suggestion
For instance, to configure the main properties of the {prod-short} server component, the Operator generates a necessary `ConfigMap`, called `che`. And any change in the `ConfigMap` automatically triggers a restart of the {prod-short} Pod.
```

File: src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+
+.Additional resources
+
+* link:https://docs.openshift.com/container-platform/latest/operators/olm-what-operators-are.html[Understanding Operators].
+
+* link:https://docs.openshift.com/container-platform/latest/operators/crds/crd-managing-resources-from-crds.html[Understanding Custom Resources].
+
+* To learn how to modify the `CheCluster` Custom Resource, see the chosen installation procedure.
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc:17]: I assume in the future you would point to a documentation reference.

File: src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc [hunk 6]
```diff
@@ -46,8 +66,8 @@ Property: Default value: Description
 `tlsSupport`: `false`: Instructs the Operator to deploy {prod-short} in TLS mode. Enabling TLS requires enabling the `selfSignedCert` field.
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc:66]: We have to check that this list doesn't miss some of the new fields (by checking in the operator code)

File: src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc [hunk 7]
```diff
@@ -113,8 +132,8 @@ Property: Default value: Description
 `tlsSecretName`: omit: Name of a secret that is used to set ingress TLS termination if TLS is enabled. See also the `tlsSupport` field.
 :===
 
-== `installation` defines the observed state of {prod-short} installation
-
+[id="checluster-custom-resource-installation-settings_{context}"]
+.`CheCluster` Custom Resource `installation` defines the observed state of {prod-short} installation
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc]: Its not `installation`, it's the `status` of the `CheCluster` custom resource:
```
spec:
    auth:
        ...
    server:
        ...
status:
    cheClusterRunning: xxx
    cheUrl: xxx
    ...
```
```

File: src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc [hunk 8]
```diff
@@ -132,37 +151,3 @@ Property: Description
 `pluginRegistryURL`: Public URL to the Plugin registry.
 `reason`: A brief CamelCase message with details about why the Pod is in this state.
 :===
-
```
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc:135]: Where was this part moved ?

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc [hunk 9]
```diff
@@ -14,19 +14,22 @@ This section describes how to install {prod-short} on OpenShift 3 with the CLI m
 * The `oc` OpenShift 3.11 CLI management tool is installed and configured. See link:https://docs.openshift.com/container-platform/3.11/cli_reference/get_started_cli.html#installing-the-cli[Installing the OpenShift 3.11 CLI].
 * The `{prod-cli}` CLI management tool is installed. See link:{site-baseurl}che-7/installing-the-{prod-cli}-management-tool/[Installing the `{prod-cli}` CLI management tool].
 * The URL for applications on this OpenShift instance is known as `<OPENSHIFT_APPS_URL>`. See xref:preparing-openshift-3-for-installing-che_{context}[].
+* Prepare a YAML file `configuration.yaml` containing the configuration options. See link:{site-baseurl}che-7/configuring-the-che-installation[Configuring the {prod-short} installation].
```
Comment (davidfestal) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc]: I would make this step optional. Only if the user wants to use settings that cannot be set directly by the main chectl comand line parameters.

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc [hunk 10]
```diff
@@ -15,6 +15,7 @@ This section describes how to install {prod-short} on OpenShift 3 with the CLI m
 * The `oc` OpenShift 3.11 CLI management tool is installed and configured. See link:https://docs.openshift.com/container-platform/3.11/cli_reference/get_started_cli.html#installing-the-cli[Installing the OpenShift 3.11 CLI].
 * The `{prod-cli}` CLI management tool is installed. See link:{site-baseurl}che-7/installing-the-{prod-cli}-management-tool/[Installing the `{prod-cli}` CLI management tool].
 * The URL for applications on this OpenShift instance is known as `<OPENSHIFT_APPS_URL>`. See xref:preparing-openshift-3-for-installing-che_{context}[].
+* Prepare a YAML file `configuration.yaml` containing the configuration options. See link:{site-baseurl}che-7/configuring-the-che-installation[Configuring the {prod-short} installation].
```
Comment (davidfestal) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc]: I would make this step optional. Only if the user wants to use settings that cannot be set directly by the main chectl comand line parameters.

File: src/main/pages/che-7/overview/proc_installing-che-using-the-che-operator-in-openshift-4-web-console.adoc [hunk 11]
```diff
@@ -26,9 +26,8 @@ This section describes how to install {prod-short} using the {prod-short} Operat
 
 . Click the *Create Instance* link in *Provided APIs* section.
 
-. The *Create {prod-short} Cluster* page is displayed.
-
-. Leave the default values as they are.
+. The *Create CheCluster* page contains the configuration of the {prod-short} cluster to create. It is a YAML document describing the `CheCluster` Custom Resource to create. For an installation using the default configuration, keep the default values.
```
Comment (davidfestal) [inline src/main/pages/che-7/overview/proc_installing-che-using-the-che-operator-in-openshift-4-web-console.adoc]:
```suggestion
. The *Create CheCluster* page contains the configuration of the overall {prod-short} installation to create. It is a YAML fragment describing the `CheCluster` Custom Resource to create. For an installation using the default configuration, keep the default values.
```

**Review by davidfestal — COMMENTED** [review]
I assume we should add a reference to the `CheCluster`-based configuration from the installation based on chectl (at least when using the `operator` installer). A bit like what you did for the Openshift 3 installation procedure.

File: src/main/pages/che-7/installation-guide/assembly_configuring-che-overview.adoc (hunk 1 above)
Comment (themr0c) [reply src/main/pages/che-7/installation-guide/assembly_configuring-che-overview.adoc:23]: Yes, but one thing at a time, this PR is already huge!

File: src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc (hunk 5 above)
Comment (themr0c) [reply src/main/pages/che-7/installation-guide/con_understanding-the-checluster-custom-resource.adoc:17]: Yes, this PR makes visible some missing parts elsewhere, but let's keep the current PR in a reasonable scope.

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc (hunk 10 above)
Comment (themr0c) [reply src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc]:
```suggestion
*  To apply settings that can't be set directly by the main {prod-cli} command line parameters, prepare a YAML file `configuration.yaml` containing the configuration options. See link:{site-baseurl}che-7/configuring-the-che-installation[Configuring the {prod-short} installation].
```

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc (hunk 9 above)
Comment (themr0c) [reply src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc]:
```suggestion
* To apply settings that can't be set directly by the main {prod-cli} command line parameters, prepare a YAML file `configuration.yaml` containing the configuration options. See link:{site-baseurl}che-7/configuring-the-che-installation[Configuring the {prod-short} installation].
```

File: src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc (hunk 7 above)
Comment (themr0c) [reply src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc]:
```suggestion
.`CheCluster` Custom Resource `status` defines the observed state of {prod-short} installation
```

File: src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc (hunk 8 above)
Comment (themr0c) [reply src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc:135]: src/main/pages/che-7/installation-guide/ref_che-configmaps-fields-reference.adoc → src/main/pages/che-7/installation-guide/ref_checluster-custom-resource-fields-reference.adoc

**Review by MichalMaler — APPROVED** [review]

**Pull request merged**